crates/aranya-aqc-util/src/handler.rs

```diff
@@ -0,0 +1,421 @@
+        }
+        Ok(psk)
+    }
+
+    /// Converts a [`UniPeerEncap`] into a PSK.
+    pub fn uni_channel_received<E: Engine>(
+        &mut self,
+        eng: &mut E,
+        effect: &UniChannelReceived<'_>,
+    ) -> Result<UniPsk<E::CS>, Error> {
+        if (self.device_id != effect.send_id && self.device_id != effect.recv_id)
+            || self.device_id == effect.author_id
+        {
+            return Err(Error::NotRecipient);
+        }
+
+        let encap =
+            UniPeerEncap::from_bytes(effect.encap).map_err(|err| Error::Crypto(err.into()))?;
+
+        let our_sk = &self
+            .store
+            .get_key(eng, effect.peer_enc_key_id.into())
+            .map_err(|_| Error::KeyStore)?
+            .ok_or(Error::KeyNotFound("device encryption key"))?;
+        let their_pk = &decode_enc_pk(effect.author_enc_pk).map_err(|err| {
+            error!("unable to decode `EncryptionPublicKey`: {err}");
+            Error::Encoding
+        })?;
+        let ch = UniChannel {
+            psk_length_in_bytes: effect.psk_length_in_bytes,
+            parent_cmd_id: effect.parent_cmd_id,
+            seal_id: effect.send_id,
+            open_id: effect.recv_id,
+            our_sk,
+            their_pk,
+            label: effect.label_id.into(),
+        };
+
+        let psk = if self.device_id == effect.send_id {
+            UniSendPsk::from_peer_encap(&ch, encap)
+                .inspect_err(|err| {
+                    error!(?err, "unable to derive uni send PSK from peer encap");
+                })
+                .map(UniPsk::SendOnly)?
+        } else {
+            UniRecvPsk::from_peer_encap(&ch, encap)
+                .inspect_err(|err| {
+                    error!(?err, "unable to derive uni recv PSK from peer encap");
+                })
+                .map(UniPsk::RecvOnly)?
+        };
+        if psk.identity() != effect.channel_id {
+            bug!("PSK identity does not match `effect.channel_id`");
+        }
+        Ok(psk)
+    }
+}
+
+/// Data from the `AqcUniChannelCreated` effect.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct UniChannelCreated<'a> {
+    /// Uniquely identifies the channel.
+    pub channel_id: UniChannelId,
+    /// The unique ID of the previous command.
+    pub parent_cmd_id: Id,
+    /// The channel author's device ID.
+    pub author_id: DeviceId,
+    /// The device ID of the Device that can send messages.
+    pub send_id: DeviceId,
+    /// The device ID of the Device that can receive messages.
+    pub recv_id: DeviceId,
+    /// The channel author's encryption key ID.
+    pub author_enc_key_id: EncryptionKeyId,
+    /// The channel peer's encoded [`aranya_crypto::EncryptionPublicKey`].
+    pub peer_enc_pk: &'a [u8],
+    /// The AQC channel label.
+    pub label_id: LabelId,
+    /// A unique ID that the author can use to look up the
+    /// channel's secrets in the keystore.
+    pub author_secrets_id: BidiAuthorSecretId,
```

Comment:
Since this is in the `UniChannelCreated` struct, I believe `UniAuthorSecretId` would be the right type for this field.
```suggestion
    pub author_secrets_id: UniAuthorSecretId,
```